packages/@aws-cdk/aws-servicecatalog/README.md

```diff
@@ -9,8 +9,262 @@
+Once resources are deployed end users will be able to access them via the console or service catalog cli. 
+
+```ts
+import * as iam from '@aws-cdk/aws-servicecatalog';
+
+const portfolio = new servicecatalog.Portfolio(this, 'MyFirstPortfolio', {
+  displayName: 'MyFirstPortfolioName', 
+  providerName: 'MyTeam'
+});
+
+const user = new iam.User(this, 'MyUser', {});
+
+const role = new iam.Role(this, 'MyRole', {});
+
+const group = new iam.Group(this, 'MyGroup', {});
+
+portfolio.giveAccessToUser(user);
+portfolio.giveAccessToGroup(group);
+portfolio.giveAccessToRole(role);
+```
+
+## Constraints
+
+Constraints define governance mechanisms that allow you to manage permissions and notifications relate to actions on your stacks. 
+Using the cdk, if you do not explicity associate a product to a portfolio and add a constraint, it will automatically add the association for you. 
+
+There are rules around plurariliites for constraint configurations.  
+For example, you may not have more than one `Launch Role` or `Stack Set` constraint applied to a portfolio-product, and you may not have both of them on same assocation.  
+If a misconfigured constraint is added, `synth` will fail with an error message detailing issue.
+
+Read more at [Constraints](https://docs.aws.amazon.com/servicecatalog/latest/adminguide/constraints.html)
+
+
+### Event Notifications
+
+An event update constraint allows user to specify a list of AWS `SNS` topics to receive notifications about stack events.
+
+```ts
+import * as sns from '@aws-cdk/aws-sns';
+
+const portfolio = new servicecatalog.Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+});
+
+const product = new servicecatalog.Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+const topics = [new sns.Topic(stack, 'topic1'), new sns.Topic(stack, 'topic2'), new sns.Topic(stack, 'topic3')];
+
+portfolio.addEventNotifications({ topics: topics, product: product });
+```
+
+### Launch Role
+
+A launch role constraint specifies a role that the end user assumes when launching a product. You cannot have more than one of these applied to a portfolio-product association.
+
+```ts
+const portfolio = new servicecatalog.Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+});
+
+const product = new servicecatalog.Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+const launchRole = new iam.Role(this, 'LaunchRole', {
+  assumedBy: new iam.AccountRootPrincipal(),
+});
+
+portfolio.addLaunchRole({
+  role: launchRole,
+  product: product,
+});
```

Comment:
Same comment here - how about 

```suggestion
portfolio.addLaunchRole(product, launchRole);
```

?

Reply:
Started going down this rabbit hole and modifying other constraints - I'm not sure the tradeoff between simplicity and readability.  Having the options like this makes it very clear to end user what args are going where and you can immediately tell by looking at the code.